### PR TITLE
Server: tags and accumulator

### PR DESCRIPTION
**Pull request by hsidd1**

### CODE REVIEW

**Comment by hsidd1** [comment]
We could clean out the deprecated files if needed. just left it here for a raw merge, but also there'll be a record of it somewhere in the repo history

File: results/server/tags_validation.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            if value["type"] == "bool":
+                if value["greaterThan"] not in [0, 1]:
+                    raise ValueError("greaterThan value must be 0 or 1")
+                if value["lessThan"] not in [0, 1]:
+                    raise ValueError("lessThan value must be 0 or 1")
+            if value["type"] == "string":
+                if not isinstance(value["greaterThan"], str):
+                    raise ValueError("greaterThan value must be a string")
+                if not isinstance(value["lessThan"], str):
+                    raise ValueError("lessThan value must be a string")
```
Comment (TylerStAmour) [inline results/server/tags_validation.py:55]: Shouldn't string comparisons only have an expectedval?

File: results/server/tags_validation.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    @staticmethod
+    def validate_tags(tags: dict):
+        for key, value in tags.items():
+            if value["greaterThan"] > value["lessThan"]:
+                raise ValueError("greaterThan value must be less than lessThan value")
+            if value["type"] not in ["float", "int", "bool", "string"]:
+                raise ValueError("type must be float, int, bool or string")
+
+    @staticmethod
+    def validate_tags2(tags):
```
Comment (TylerStAmour) [inline results/server/tags_validation.py:36]: Is there a need for two validate_tags functions or is this part of the deprecated files?

**Review by TylerStAmour — APPROVED** [review]

File: results/server/tags_validation.py (hunk 1 above)
Comment (hsidd1) [reply results/server/tags_validation.py:55]: Validation is done with schema, this one is deprecated we can remove it

File: results/server/proto/results_pb2.pyi [hunk 3]
```diff
@@ -17,7 +17,14 @@ class SubmitTagRequest(_message.Message):
     value_int: int
     value_float: float
     value_bool: bool
-    def __init__(self, tag: _Optional[str] = ..., value_str: _Optional[str] = ..., value_int: _Optional[int] = ..., value_float: _Optional[float] = ..., value_bool: bool = ...) -> None: ...
+    def __init__(
```
[the remark this answers is not in the document]
Comment (hsidd1) [reply results/server/proto/results_pb2.pyi:20]: Yea that was just a pep8 formatter I ran I didn't acc do anything here

File: results/server/tags_validation.py (hunk 2 above)
Comment (hsidd1) [reply results/server/tags_validation.py:36]: The latter

**Pull request merged**